diff -rupN buildroot_tst/Makefile buildroot-2015.08.01/Makefile
--- buildroot_tst/Makefile	2017-05-31 00:00:39.745257182 -0400
+++ buildroot-2015.08.01/Makefile	2016-06-04 07:13:51.506550902 -0400
@@ -104,7 +104,7 @@ noconfig_targets := menuconfig nconfig g
 # something else than one of the nobuild_targets.
 nobuild_targets := source source-check \
 	legal-info external-deps _external-deps \
-	clean distclean
+	clean clean_all distclean
 ifeq ($(MAKECMDGOALS),)
 BR_BUILDING = y
 else ifneq ($(filter-out $(nobuild_targets),$(MAKECMDGOALS)),)
@@ -432,7 +432,7 @@ prepare: $(BUILD_DIR)/buildroot-config/a
 
 world: target-post-image
 
-.PHONY: all world toolchain dirs clean distclean source outputmakefile \
+.PHONY: all world toolchain dirs clean clean_all distclean source outputmakefile \
 	legal-info legal-info-prepare legal-info-clean printvars help \
 	list-defconfigs target-finalize target-post-image source-check
 
@@ -822,11 +822,23 @@ printvars:
 				$(origin $V)), \
 		$(info $V=$($V) ($(value $V)))))
 
-clean:
+clean_all:
 	rm -rf $(TARGET_DIR) $(BINARIES_DIR) $(HOST_DIR) \
 		$(BUILD_DIR) $(BASE_DIR)/staging \
 		$(LEGAL_INFO_DIR) $(GRAPHS_DIR)
 
+clean:
+	@echo "Removing" $(TARGET_DIR)
+	@echo "Removing" $(BINARIES_DIR)
+	@echo "Removing" $(BASE_DIR)/staging
+	@echo "Removing" $(LEGAL_INFO_DIR)
+	@echo "Removing" $(GRAPHS_DIR)
+	@rm -rf $(TARGET_DIR) $(BINARIES_DIR) $(BASE_DIR)/staging $(LEGAL_INFO_DIR) $(GRAPHS_DIR)
+	rm -f $(O)/build/uboot-custom/.stamp_configured
+	rm -f $(O)/build/uboot-custom/.stamp_built $(O)/build/linux-custom/.stamp_built $(O)/build/*/.stamp_target_installed
+	cd $(O)/build/uboot-custom && $(MAKE) mrproper
+	cd $(O)/build/linux-custom && $(MAKE) clean
+
 distclean: clean
 ifeq ($(DL_DIR),$(TOPDIR)/dl)
 	rm -rf $(DL_DIR)
@@ -839,7 +851,8 @@ endif
 
 help:
 	@echo 'Cleaning:'
-	@echo '  clean                  - delete all files created by build'
+	@echo '  clean_all              - delete all files created by build'
+	@echo '  clean                  - delete all object files created by build'
 	@echo '  distclean              - delete all non-source files (including .config)'
 	@echo
 	@echo 'Build:'
